src/zarr/core/group.py

```diff
@@ -214,13 +341,51 @@ async def getitem(
         store_path = self.store_path / key
         logger.debug("key=%s, store_path=%s", key, store_path)
 
+        # Consolidated metadata lets us avoid some I/O operations so try that first.
+        if self.metadata.consolidated_metadata is not None:
```

Comment:
Consider splitting this into a `_getitem_consolidated` and `_getitem_unconsolidated`?